.github/workflows/weekly_tests.yaml

```diff
@@ -6,410 +6,142 @@ on:
-            failed_tests+="[charm-prometheus-libvirt-exporter](https://github.com/canonical/charm-prometheus-libvirt-exporter/actions), "
-          fi
-          if [ "${{ steps.charm-simple-streams.outcome }}" != "success" ]; then
-            failed_tests+="[charm-simple-streams](https://github.com/canonical/charm-simple-streams/actions), "
-          fi
-          if [ "${{ steps.charm-storage-connector.outcome }}" != "success" ]; then
-            failed_tests+="[charm-storage-connector](https://github.com/canonical/charm-storage-connector/actions), "
-          fi
-          if [ "${{ steps.charm-sysconfig.outcome }}" != "success" ]; then
-            failed_tests+="[charm-sysconfig](https://github.com/canonical/charm-sysconfig/actions), "
-          fi
-          if [ "${{ steps.charm-userdir-ldap.outcome }}" != "success" ]; then
-            failed_tests+="[charm-userdir-ldap](https://github.com/canonical/charm-userdir-ldap/actions), "
-          fi
-          if [ "${{ steps.prometheus-hardware-exporter.outcome }}" != "success" ]; then
-            failed_tests+="[prometheus-hardware-exporter](https://github.com/canonical/prometheus-hardware-exporter/actions), "
-          fi
-          if [ "${{ steps.prometheus-juju-exporter.outcome }}" != "success" ]; then
-            failed_tests+="[prometheus-juju-exporter](https://github.com/canonical/prometheus-juju-exporter/actions), "
-          fi
-          if [ "${{ steps.prometheus-juju-backup-all-exporter.outcome }}" != "success" ]; then
-            failed_tests+="[prometheus-juju-backup-all-exporter](https://github.com/canonical/prometheus-juju-backup-all-exporter/actions), "
-          fi
-          if [ "${{ steps.snap-tempest.outcome }}" != "success" ]; then
-            failed_tests+="[snap-tempest](https://github.com/canonical/snap-tempest/actions), "
-          fi
-          if [ "${{ steps.charmed-openstack-upgrader.outcome }}" != "success" ]; then
-            failed_tests+="[charmed-openstack-upgrader](https://github.com/canonical/charmed-openstack-upgrader/actions), "
-          fi
-          # Set the output variable to pass the failed tests to the next job
-          echo "failed_tests=$failed_tests" >> "$GITHUB_OUTPUT"
-
-  notify-on-failure:
-    needs: [tests-1, tests-2]
-    if: ${{ failure() }}
-    name: Notify Mattermost Channel
+          result=$(cat <<-END
+              {
+                  "job-index": "${{ strategy.job-index }}",
+                  "repo": "${{ matrix.repo }}",
+                  "branch": "${{ matrix.branch }}",
+                  "workflow_file_name":"${{ matrix.workflow_file_name }}",
+                  "conclusion": "${{ steps.dispatched-tests.outcome }}",
+                  "workflow_url": "${{ steps.dispatched-tests.outputs.workflow_url }}"
+              }
+          END
+          )
+          echo $result > "result-${{ strategy.job-index }}.json"
```

Comment:
non-blocking: It would be better to use repo+workflow name instead of `strategy.job-index` so a human can identify a file quickly when troubleshooting

Reply:
We can use all parameter from matrix, but these artifacts are only for internal usage, and you should found the same information in each job, so I do not see reason for anyone to go there.

Reply:
I still think it would be better having a more human-readable filename, but this is not at all a blocker. Resolving.